mmv1/third_party/terraform/resources/resource_container_node_pool.go.erb

```diff
@@ -302,10 +302,22 @@ func resourceContainerNodePoolCreate(d *schema.ResourceData, meta interface{}) e
 	timeout := d.Timeout(schema.TimeoutCreate)
 	startTime := time.Now()
 
-	// Set the ID before we attempt to create - that way, if we receive an error but
-	// the resource is created anyway, it will be refreshed on the next call to
-	// apply.
-	d.SetId(fmt.Sprintf("projects/%s/locations/%s/clusters/%s/nodePools/%s", nodePoolInfo.project, nodePoolInfo.location, nodePoolInfo.cluster, nodePool.Name))
+	var id = fmt.Sprintf("projects/%s/locations/%s/clusters/%s/nodePools/%s", nodePoolInfo.project, nodePoolInfo.location, nodePoolInfo.cluster, nodePool.Name)
+	name := getNodePoolName(id)
+
+	clusterNodePoolsGetCall := config.NewContainerBetaClient(userAgent).Projects.Locations.Clusters.NodePools.Get(nodePoolInfo.fullyQualifiedName(name))
```

Comment:
nit: should we comment above here as well, to explain why we're doing a prefetch?

Reply:
done